pkg/server/backuprestoreserver.go

```diff
@@ -151,19 +153,47 @@ func (b *BackupRestoreServer) runServerWithSnapshotter(ctx context.Context, rest
 	handler := b.startHTTPServer(etcdInitializer, ssr)
 	defer handler.Stop()
 
-	ssrStopCh := make(chan struct{})
+	leaderCallbacks := &leaderelection.LeaderCallbacks{
+		OnStartedLeading: func(ctx context.Context) {
+			// Get the updated snapshotter state
+			err := ssr.GetLatestSnapshotterState()
+			if err != nil {
+				fmt.Printf("Unable to get Latest SnapshotterState: %v", err)
+			}
+			handler.Snapshotter = ssr
+		},
+		OnStoppedLeading: func() {
+			// stops the running snapshotter
+			ssr.SsrStateMutex.Lock()
+			if ssr.SsrState == brtypes.SnapshotterActive {
+				ssr.SsrStateMutex.Unlock()
+				ssrStopCh <- emptyStruct
+			} else {
+				ssr.SsrStateMutex.Unlock()
+			}
```

Comment:
```suggestion
			ssr.SsrStateMutex.Lock()
			ssr.SsrStateMutex.Unlock()
			if ssr.SsrState == brtypes.SnapshotterActive {
				ssrStopCh <- emptyStruct
			}
```

This might inactivate the snapshotter but what happens if some other goroutine activates it (by toggling using the same `sstStopCh`) before the leader election has been won?